modules/core/nuxeo-core-event/src/test/java/org/nuxeo/ecm/core/event/test/TestEventServiceComponent.java

```diff
@@ -107,8 +111,9 @@ public void testAsync() throws InterruptedException {
         Event test2 = new EventImpl("testnotmached", new EventContextImpl()); // NOSONAR
         test2.setIsCommitEvent(true);
         eventService.fireEvent(test2);
+        txFeature.nextTransaction();
         eventService.waitForAsyncCompletion();
```

Comment:
can we remove this?

Reply:
No, we can't, we need to wait for the async listeners. `TransactionalFeature#nextTransaction` wait for them only if we are using the WorkManager.